pkg/healthchecker/health_checker.go

```diff
@@ -102,10 +98,21 @@ func (hc *healthChecker) CheckHealth() (bool, error) {
 
 // logPatternHealthCheck checks for the provided logPattern occurrences in the service logs.
 // Returns true if the pattern is empty or does not exist logThresholdCount times since start of service, false otherwise.
-func logPatternHealthCheck(service, logStartTime string, logPatternsToCheck map[string]int) (bool, error) {
+func logPatternHealthCheck(service string, loopBackTime time.Duration, logPatternsToCheck map[string]int) (bool, error) {
 	if len(logPatternsToCheck) == 0 {
 		return true, nil
 	}
+	uptimeFunc := getUptimeFunc(service)
+	uptime, err := uptimeFunc()
+	if err != nil {
+		glog.Warningf("Failed to get the uptime: %+v", err)
+		return true, err
+	}
+
+	logStartTime := time.Now().Add(-uptime).Format(types.LogParsingTimeLayout)
```

Comment:
For more predictability capture `time.Now()` at 1 point above, probably before the `uptimeFunc()` call.

Reply:
the logStartTime needs the uptime so it cannot go before uptimeFunc() call